### PR TITLE
feat: add windows recording rules to azure monitor metrics addon (managed prometheus) 

### PR DESCRIPTION
**Pull request by bragi92**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/aks-preview/azext_aks_preview/_help.py [hunk 1]
```diff
@@ -907,6 +907,9 @@
         - name: --grafana-resource-id
           type: string
           short-summary: Resource ID of the Azure Managed Grafana Workspace
+        - name: --grafana-resource-id
+          type: bool
+          short-summary: Enable Windows Recording Rules when enabling the Azure Monitor Metrics addon
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/_help.py:912]: wrong name? should be --enable-windows-recording-rules, right?

**Review by FumingZhang — COMMENTED** [review]
Queued a [pipeline](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=70120293&view=results) to validate the modified test case `test_aks_update_with_azuremonitormetrics`, though CI is already failing on that one..

File: src/aks-preview/azext_aks_preview/_help.py (hunk 1 above)
Comment (bragi92) [reply src/aks-preview/azext_aks_preview/_help.py]: yes! updated.

File: src/aks-preview/azext_aks_preview/azuremonitorprofile.py [hunk 2]
```diff
@@ -508,20 +507,8 @@ def link_grafana_instance(cmd, raw_parameters, azure_monitor_workspace_resource_
     return GrafanaLink.SUCCESS
 
 
-def create_rules(cmd, cluster_subscription, cluster_resource_group_name, cluster_name, azure_monitor_workspace_resource_id, mac_region):
+def put_rules(cmd, default_rule_group_id, default_rule_group_name, mac_region, azure_monitor_workspace_resource_id, cluster_name, default_rules_template, url, i):
```
Comment (bragi92) [inline src/aks-preview/azext_aks_preview/azuremonitorprofile.py:510]: also enable the windows rules

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**